### PR TITLE
Thermal validation and loading [ANT-1213]

### PR DESCRIPTION
**Pull request by payetvin**
Modular loading: separating loading and validation functions

### CODE REVIEW

File: src/libs/antares/study/parts/thermal/cluster_list.cpp [hunk 1]
```diff
@@ -159,52 +159,57 @@ bool ThermalClusterList::loadFromFolder(Study& study, const AnyString& folder, A
...
+
+
+bool ThermalClusterList::validateClusters(const Parameters& parameters) const
+{
+    bool ret = true;
+
+    for (const auto& cluster : allClusters_)
+    {
+        // update the minUpDownTime
+        cluster->minUpDownTime = std::max(cluster->minUpTime, cluster->minDownTime);
```
Comment (JasonMarechal25) [inline src/libs/antares/study/parts/thermal/cluster_list.cpp:188]: Need a bit of clarification, maybe more details in the comment. It's strange to see a std::max on a minX value.
Let's add information on why we're doing this

File: src/libs/antares/study/parts/thermal/cluster_list.cpp [hunk 2]
```diff
@@ -606,18 +585,31 @@ bool ThermalClusterList::saveEconomicCosts(const AnyString& folder) const
...
+{
+    auto hasPrepro = [](auto c) { return (bool)c->prepro; };
+
+    const bool globalThermalTSgeneration = study.parameters.timeSeriesToGenerate
+                                           & timeSeriesThermal;
+
+    return std::ranges::all_of(allClusters_ | std::views::filter(hasPrepro),
+            [&study, &globalThermalTSgeneration](auto& c)
+    {
+        bool result = true;
```
Comment (JasonMarechal25) [inline src/libs/antares/study/parts/thermal/cluster_list.cpp]: if NOT (study.usedByTheSolver && globalThermalTSgeneration)  that is to say if !study.usedByTheSolver or !globalThermalTSgeneration result will be true. Moreover it is constant for all element.
It seems that we can return true before the computation of all_of if either value is false.

**Review by JasonMarechal25 — CHANGES_REQUESTED** [review]

File: src/libs/antares/study/parts/thermal/cluster_list.cpp (hunk 1 above)
Comment (payetvin) [reply src/libs/antares/study/parts/thermal/cluster_list.cpp:188]:
```
    //! Max entre . minUp/minDown time (1..168)
    uint minUpDownTime;
```
The naming isn't great, can be changed to maxMinUpMinDownTime

File: src/libs/antares/study/parts/thermal/cluster_list.cpp [hunk 3]
```diff
@@ -159,51 +159,69 @@ bool ThermalClusterList::loadFromFolder(Study& study, const AnyString& folder, A
...
+}
+
+
+bool ThermalClusterList::validateClusters(const Parameters& parameters) const
+{
+    bool ret = true;
+
+    for (const auto& cluster : allClusters_)
+    {
+        if (cluster->minUpTime < 1)
```
Comment (JasonMarechal25) [inline src/libs/antares/study/parts/thermal/cluster_list.cpp]: Please replace with: std::clamp(cluster->minUpTime, 1, 168)
And replace 1 and 168 with named constant if possible.

**Review by JasonMarechal25 — CHANGES_REQUESTED** [review]

File: src/libs/antares/study/parts/thermal/cluster_list.cpp (hunk 3 above)
Comment (JasonMarechal25) [reply src/libs/antares/study/parts/thermal/cluster_list.cpp:184]: It seems that some value "underflow". Some value stored as unint fall under 0, going back to UINT_MAX. When clamped they are clamped to 168.
At some point somewhere else we wrongly convert some value or don't check bounds. But it is outside of this PR to solve this.

**Review by JasonMarechal25 — APPROVED** [review]

**Pull request merged**